addons/product/models/product.py

```diff
@@ -378,17 +378,37 @@ def _name_get(d):
         self.check_access_rule("read")
 
         result = []
+
+        # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
+        # Use `load=False` to not call `name_get` for the `product_tmpl_id`
+        self.sudo().read(['name', 'default_code', 'product_tmpl_id', 'attribute_value_ids'], load=False)
```

Comment:
A comment why is needed to be read with sudo if is necessary please.

Reply:
Agree. Also, using sudo will prefetch the fields for the sudo() user instead of user in self, won't it?

Reply:
Sudo is used because the products are looped through as Sudo in line 400:

`for product in self.sudo():`

Reply:
I can add a comment if you want but I think it's a bit more clear when looking at the function as a whole rather than the highlighted changes. :)